apollo-runtime/src/main/java/com/apollographql/apollo/internal/interceptor/ApolloServerInterceptor.java

```diff
@@ -138,33 +138,29 @@ Call httpCall(Operation operation, CacheHeaders cacheHeaders) throws IOException
     return httpCallFactory.newCall(requestBuilder.build());
   }
 
-  private RequestBody httpRequestBody(Operation operation) throws IOException {
+  static ByteString httpRequestBody(Operation operation, ScalarTypeAdapters scalarTypeAdapters,
+      boolean writeQueryDocument) throws IOException {
     Buffer buffer = new Buffer();
     JsonWriter jsonWriter = JsonWriter.of(buffer);
     jsonWriter.setSerializeNulls(true);
     jsonWriter.beginObject();
-    if (sendOperationIdentifiers) {
-      jsonWriter.name("id").value(operation.operationId());
-    } else {
-      jsonWriter.name("query").value(operation.queryDocument().replaceAll("\\n", ""));
-    }
     jsonWriter.name("operationName").value(operation.name().name());
     jsonWriter.name("variables").beginObject();
     operation.variables().marshaller().marshal(new InputFieldJsonWriter(jsonWriter, scalarTypeAdapters));
     jsonWriter.endObject();
+    jsonWriter.name("extensions")
+        .beginObject()
+        .name("persistedQuery")
+        .beginObject()
+        .name("version").value(1)
+        .name("sha256Hash").value(operation.operationId())
+        .endObject()
+        .endObject();
+    if (writeQueryDocument) {
+      jsonWriter.name("query").value(operation.queryDocument().replaceAll("\\n", ""));
```

Comment:
Are you replacing the newlines here because operation id is not the hash of operation.queryDocument()?

Reply:
Operation ID is the hash of the query document, right? If the query document you are sending has new lines removed, won't the hash be different? Or is the hash calculated with the new lines already removed?